match/new_orleans_so.py

```diff
@@ -39,28 +40,7 @@ def deduplicate_cprr_19_personnel(cprr):
     )
     clusters = matcher.get_index_clusters_within_thresholds(decision)
     # canonicalize name and uid
-    for cluster in clusters:
-        uid, first_name, last_name, middle_initial = None, "", "", ""
-        for idx in cluster:
-            row = df.loc[idx]
-            if (
-                uid is None
-                or len(row.first_name) > len(first_name)
-                or (
-                    len(row.first_name) == len(first_name)
-                    and len(row.last_name) > len(last_name)
-                )
-            ):
-                uid, first_name, last_name, middle_initial = (
-                    idx,
-                    row.first_name,
-                    row.last_name,
-                    row.middle_initial,
-                )
-        cprr.loc[cprr.uid.isin(cluster), "uid"] = uid
-        cprr.loc[cprr.uid == uid, "first_name"] = first_name
-        cprr.loc[cprr.uid == uid, "last_name"] = last_name
-        cprr.loc[cprr.uid == uid, "middle_initial"] = middle_initial
+    canonicalize_names(cprr, clusters)
```

Comment:
Clearly we need to modify `canonicalize_names` to also accommodate `middle_initial` column, and possibly also `middle_name` column. The trick is to make sure the columns are all optional so for example if the frame only have `last_name` and `middle_initial`, it should still work without a hitch.

Reply:
Had this thought a while ago, but I forgot to bring it up. It might be worth standardizing `middle_initial `columns to `middle_name `columns as it's probable that we'll receive data that will allow us to replace initials with full middle names

Reply:
I suppose we don't lose anything if we do that. @rajivsinclair ?